katcp/test/test_server.py

```diff
@@ -325,13 +338,91 @@ def test_has_sensor(self):
         self.server.add_sensor(katcp.Sensor.boolean('blaah', 'blaah sens'))
         self.assertTrue(self.server.has_sensor('blaah'))
 
+    def test_excluded_default_handlers(self):
+        """
+        Test that default handers from higher KATCP versions are not included
+
+        """
+        # TODO NM 2017-04-18 This should probably be removed if we do official
+        # KATCP v5.1 release and make it the default server version
+        self.assertNotIn('request-timeout-hint', self.server._request_handlers)
+
 
 class test_DeviceServerAsync(test_DeviceServer):
     def setUp(self):
         super(test_DeviceServerAsync, self).setUp()
         self.server.set_concurrency_options(
             thread_safe=False, handler_thread=False)
 
+class test_DeviceServer51(test_DeviceServer):
+    """Proposed additional tests for Verion 5.1 server"""
+
+    expected_connect_messages = (
+        r'#version-connect katcp-protocol 5.1-IMT',
+        # Will have to be updated for every library version bump
+        r'#version-connect katcp-library katcp-python-%s' % katcp_version,
+        r'#version-connect katcp-device deviceapi-5.6 buildy-1.2g')
+
+    def setUp(self):
+        class DeviceTestServer51(DeviceTestServer):
+            PROTOCOL_INFO = katcp.ProtocolFlags(
+            5, 1, [
+                katcp.ProtocolFlags.MULTI_CLIENT,
+                katcp.ProtocolFlags.MESSAGE_IDS,
+                katcp.ProtocolFlags.REQUEST_TIMEOUT_HINTS])
+        self.server = DeviceTestServer51('', 0)
+        self.server.set_concurrency_options(
+            thread_safe=False, handler_thread=False)
+
+    def test_excluded_default_handlers(self):
+        pass                  #  No excluded default handlers for v5.1 as of yet
+
+    def test_request_timeout_hint(self):
+        req = mock_req('request-timeout-hint')
+        handle_mock_req(self.server, req)
+        # Check that the default test server "slow command" request has the
+        # expected timeout hint of 99 seconds
+        self._assert_msgs_equal([req.reply_msg],
+                                ['!request-timeout-hint ok 1'])
+        self._assert_msgs_equal(req.inform_msgs,
+                                ['#request-timeout-hint slow-command 99.0'])
+        # Check that a request without a hint gives a timeout of 0 if explicitly
+        # asked for
+        req = mock_req('request-timeout-hint', 'help')
+        handle_mock_req(self.server, req)
+        self._assert_msgs_equal([req.reply_msg],
+                                ['!request-timeout-hint ok 1'])
+        self._assert_msgs_equal(req.inform_msgs,
+                                ['#request-timeout-hint help 0.0'])
+
+        # Now set hint on help command and check that it is reflected Note, in
```

Comment:
Punctuation updated in pending commit. I appreciate the attention to detail / level of pedantry, whichever way you prefer to look at it :)